### ISSUE TITLE
[List and List Item] Support sorting for AT users

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-design-system#7426]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Summary

With parity of `list`, we should provide support to AT users of list when using sorting to finalize the efforts initiated in #6554.

### Actual Behavior

When active, both JAWS and NVDA are unable to sort `list-items` in the `list`.

### Expected Behavior

Same behavior as keyboard navigation without AT enabled, where `space` activates the move, and the arrow keys place the item.

### Reproduction Sample

https://codepen.io/geospatialem/pen/bGQzavG

### Reproduction Steps

1. Activate JAWS or NVDA
2. Open the [code sample](https://codepen.io/geospatialem/pen/bGQzavG)
    - JAWS: Chrome
    - NVDA: Firefox
3. Using your keyboard navigate to the first List Item's movable button
4. Press the `space` key
5. Press the down arrow key
6. Observe the breakage (_Note: NVDA takes two down arrows to break_)

### Reproduction Version

1.5.0-next.33 (1.5.0)

### Relevant Info

- List parity: https://github.com/Esri/calcite-design-system/pull/7109#pullrequestreview-1553114334
- Screen reader support: https://github.com/Esri/calcite-design-system/issues/3982#issuecomment-1206487248
- R&D efforts: https://github.com/Esri/calcite-design-system/issues/3982#issuecomment-1449143638

### Priority impact

p3 - want for upcoming milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react

### Esri team

Calcite (dev)

**Comment by geospatialem** [comment]
Effort should be considered with the effort in https://github.com/Esri/calcite-design-system/issues/7537

**Comment by geospatialem** [comment]
This also applies to the sorting with the `block` component in `2.0.0` with both Windows AT, JAWS and NVDA, as outlined in https://codepen.io/geospatialem/pen/PoVXeVL cc @driskull

**Comment by driskull** [comment]
@geospatialem what exactly is the breakage? Is it not reading something, is it not moving?

**Comment by geospatialem** [comment]
> @geospatialem what exactly is the breakage? Is it not reading something, is it not moving?

It's the same for the issue above and `block`, where the context is provided to AT, but sorting/moving is not possible when either JAWS or NVDA are active.

**Comment by driskull** [comment]
@geospatialem according to the following:

https://stackoverflow.com/questions/52261977/why-javascript-if-keycode-enter-key-or-spacebar-key-does-not-work-with-nvda

> It probably won't fire because NVDA intercepts the keys if you are in browse mode, which is normal. If you were to manually switch to focus mode (insert + space -- you'll hear a "typewriter" sound), the code should get passed through.

I can override this by setting the `calcite-handle` as a role of `application` which does seem to fix this.

Would this be an acceptable result? I will create a PR for testing.

**Comment by geospatialem** [comment]
Verified on Windows with NVDA and JAWS in `2.8.0-next.4`. 🎉
